docs/user/ppl/cmd/rex.rst

```diff
@@ -0,0 +1,199 @@
+    | amberduke@pyrami.com     | amberduke    | pyrami  | username=0-8,domain=10-15   |
+    | hattiebond@netagy.com    | hattiebond   | netagy  | username=0-9,domain=11-16   |
+    | nanettebates@quility.com | nanettebates | quility | username=0-11,domain=13-19  |
+    +--------------------------+--------------+---------+-----------------------------+
+
+
+Example 6: Complex Email Pattern
+=================================
+
+Extract comprehensive email components including top-level domain.
+
+PPL query::
+
+    os> source=accounts | rex field=email "(?<user>[a-zA-Z0-9._%+-]+)@(?<domain>[a-zA-Z0-9.-]+)\\.(?<tld>[a-zA-Z]{2,})" | fields email, user, domain, tld | head 3 ;
+    fetched rows / total rows = 3/3
+    +--------------------------+--------------+---------+-----+
+    | email                    | user         | domain  | tld |
+    |--------------------------+--------------+---------+-----|
+    | amberduke@pyrami.com     | amberduke    | pyrami  | com |
+    | hattiebond@netagy.com    | hattiebond   | netagy  | com |
+    | nanettebates@quility.com | nanettebates | quility | com |
+    +--------------------------+--------------+---------+-----+
+
+
+Example 7: Chaining Multiple rex Commands
+==========================================
+
+Extract initial letters from both first and last names.
+
+PPL query::
+
+    os> source=accounts | rex field=firstname "(?<firstinitial>^.)" | rex field=lastname "(?<lastinitial>^.)" | fields firstname, lastname, firstinitial, lastinitial | head 3 ;
+    fetched rows / total rows = 3/3
+    +-----------+----------+--------------+-------------+
+    | firstname | lastname | firstinitial | lastinitial |
+    |-----------|----------|--------------|-------------|
+    | Amber     | Duke     | A            | D           |
+    | Hattie    | Bond     | H            | B           |
+    | Nanette   | Bates    | N            | B           |
+    +-----------+----------+--------------+-------------+
+
+
+Comparison with Related Commands
+================================
+
+============================= ============ ============
+Feature                        rex          parse
+============================= ============ ============
+Pattern Type                   Java Regex   Java Regex
+Named Groups Required          Yes          Yes
+Filtering by Match             No           Yes  
+Multiple Matches               Yes          No
+Text Substitution              Yes          No
+Offset Tracking                Yes          No
+Underscores in Group Names     No           No
+============================= ============ ============
+
+
+Limitations
+===========
+
+There are several important limitations with the rex command:
+
+**Named Capture Group Naming:**
+
+- Named capture groups cannot contain underscores due to Java regex limitations
+- Group names must start with a letter and contain only letters and digits
+- Use ``(?<username>...)`` not ``(?<user_name>...)``
```

Comment:
remove it, add as example to explain groups can't contain understore.

Reply:
fixed